cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConfiguration.java

```diff
@@ -160,5 +164,19 @@ public boolean anyReadOnly(KafkaVersion kafkaVersion) {
         return result;
     }
 
+    /* test */
+    public static Object getDefaultValueOfProperty(String property, KafkaVersion kafkaVersion) {
+        Map<String, ConfigModel> c = readConfigModel(kafkaVersion);
+        ConfigModel optProp = c.get(property);
+        if (optProp != null) {
+            return optProp.getDefaultValue();
+        } else {
+            return null;
+        }
+    }
+
+    public boolean containsListenersChange() {
+        return asOrderedProperties().asMap().keySet().contains("listeners");
```

Comment:
No, it is comparing config from kafka broker and the generated one from configMap.

Reply:
Looking at the `io.strimzi.operator.cluster.model.KafkaBrokerConfigurationBuilder#withListeners` i believe `advertised.listeners` are changed only when `listeners` are. So it should be sufficient, should not it?

Reply:
I don't think this is true. For example change to the advertised listeners overrides in external listener would for sure change only the advertised listeners field but not the actual listener.

Reply:
That said, it seems to now match `listeners` even in the middle - so maybe you fixed this?

Reply:
If we use override, the pod is rolled because of cert has changed, isn't it?

Reply:
Yeah, you are probably right. But that would still be worth checking here. But I later realized something else - the `advertised.listeners` contains just the placeholders. So the overrides will change in the file in the config map and no there probably.

Anyway - with `contains(...)` it should now match even the advertised listeners, or? So IMHO it should be ok. I guess originaly it might have been `startsWith(...)`?